### PR TITLE
Fix #78 and #79

### LINKED ISSUE #78: Duplicate points in X warning in GMM
Warning message when running ampycloud on 2022.01.02 at 00:20 in Geneva:
`
python3.8/site-packages/sklearn/mixture/_base.py:143: ConvergenceWarning: Number of distinct clusters (2) found smaller than n_clusters (3). Possibly due to duplicate points in X.
  cluster.KMeans(
python3.8/site-packages/numpy/core/fromnumeric.py:3474: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
python3.8/site-packages/numpy/core/_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
`
**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Ubuntu 20
 - ampycloud version 0.4.0.dev0
 - Python version 3.8.10

### LINKED ISSUE #79: Add ref to docstring of frac2okta()
**Describe the change**
It would be useful to add a link to the article of [Boers et al. (2010)](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2010JD014661) to the `frac2okta()` docstring in `wmo.py`, as a reference for the conversion scheme where the 1 and 7 okta bins are larger than the others by 50%.

### PR DESCRIPTION
**Pull request by fpavogt**
**Description:**

This PR introduces a new keyword to `ncomp_from_gmm()` that allows to set the maximum number of Gaussian components to look for. This, in turn, is used to restrict it to 2 when the data is comprised of only 2 distinct numerical values for the altitude. 

This PR also cleans up several docstrings, for compliance with the contributing guidelines.

**Error(s) fixed:**

Fixes #78, fixes #79.

**Checklists**:
- [x] New code includes dedicated tests.
- [x] New code has been linted.
- [x] New code follows the project's style.
- [x] New code is compatible with the 3-Clause BSD license.
- [x] CHANGELOG has been updated.
- [ ] AUTHORS has been updated.
- [ ] Copyright years in module docstrings have been updated.

### CODE REVIEW

**Comment by nestabur** [comment]
> Hey @loforest: this PR is ready for review, whenever you have the time.
> 
> Once this is inside the `develop` branch, we need to decide when to create a new release. The mechanisms are all [automated](test/ampycloud/test_layer.py), so it is easy to do it anytime.
> 
> @nestabur: have you encountered any more issues over the past few weeks ?

No issues in the past weeks, we wait for the new release to update our service.

**Comment by fpavogt** [comment]
Thanks @nestabur for the confirmation. We will release this once this PR has been reviewed. Will let you know.

File: src/ampycloud/wmo.py [hunk 1]
```diff
@@ -169,17 +176,19 @@ def alt2code(val: Union[int, float]) -> str:
...
-    Reference: *Aerodrome Reports and Forecasts, A Users' Handbook to the Codes*,
-    WMO-No.782, 2020 edition. https://library.wmo.int/?lvl=notice_display&id=716
-
     Args:
         val (int, float): the altitude to convert, in feet.
 
     Returns:
         str: the corresponding METAR code chunk
+
+    Below 10'000 ft, the value is floored to the nearest 100 ft. Above 10'000 ft, the value is
```
Comment (loforest) [inline src/ampycloud/wmo.py:185]: How do we handle the variable resolution of the Met Report?
See GM1 MET.TR.200(a): 0-250 ft --> 50 ft; 300-10000 ft --> 100 ft
The alt2code function can only be used for AutoMETAR formatting. 
For local Met Report we need to pass the unconverted altitudes, which I assume is already done in autometpy.
We could add a comment to this function.

**Review by loforest — CHANGES_REQUESTED** [review]
Review done. I only have one small comment which can likely be solved by explanding the docstring. Otherwise, it is ready for merging. Thanks!

File: src/ampycloud/wmo.py (hunk 1 above)
Comment (fpavogt) [reply src/ampycloud/wmo.py:185]: If my memory serves me well, the autometpy calculator directly extracts the layer altitudes, NOT any WMO/ICAO-formatted codes. @nestabur would know for certain. I believe this was done precisely to keep all the operationally-relevant regulations-related formatting handled by a single software (i.e. SMART).

I will add a warning to this docstring to mention that the routine does not (currently) allow to implement EASA's rule AMC1 MET.TR.205(e)(3) and set the resolution to 50ft up to 300 ft for aerodromes with established low-visibility procedures.

**Pull request merged**